File: Lv.1/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    
    int minimum=min(n,m); //더 작은 수 찾기
    int gcd=0;
    int lcm=0;
    for(int i=1;i<=minimum;i++)
    {
        if(n%i==0 && m%i==0)
        {
            gcd=i;
        }
    }
    
    lcm=gcd*(n/gcd)*(m/gcd);
    answer.push_back(gcd);
    answer.push_back(lcm);
    return answer;
}

File: Java/LV1/최대공약수와 최소공배수.java
import java.util.*;

class Solution {
    
    public int getGCD(int a, int b){
        if(b==0){
            return a;
        }
        
        else{
            return getGCD(b, a%b);
        }
    }
    
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        
        int max = Math.max(n,m);
        int min = Math.min(n,m);
        
        int gcd = getGCD(max,min);
        answer[0]=gcd;
        answer[1]=n*m/gcd;
        
        return answer;
    }
}
